plugins/kernel_v2/config/trigger.rb

```diff
@@ -44,7 +44,9 @@ def before(*command, &block)
         command.flatten!
         blk = block
 
-        if !block_given? && command.last.is_a?(Hash)
+        if block_given? && command.last.is_a?(Hash)
+          extra_cfg = command.pop
+        elsif !block_given? && command.last.is_a?(Hash)
```

Comment:
I think for logic clarity here nesting the conditional would make it a bit clearer:

```ruby
if command.last.is_a?(Hash)
  if block_given?
    ...
  else
    ...
  end
end
```